nxc/modules/schtask.py

```diff
@@ -0,0 +1,274 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+import os
+from impacket.dcerpc.v5 import tsch, transport
+from impacket.dcerpc.v5.dtypes import NULL
+from impacket.dcerpc.v5.rpcrt import RPC_C_AUTHN_GSS_NEGOTIATE, RPC_C_AUTHN_LEVEL_PKT_PRIVACY
+from nxc.helpers.misc import gen_random_string
+from time import sleep
+
+class NXCModule:
+    """
+    Execute a scheduled task remotely as a already connected user by @Defte_
+    """
+
+    def options(self, context, module_options):
+        """
+        CMD            Command to execute
+        USER           User to execute the command as
+        """
+
+        self.cmd = self.user = self.time = None
+        if "CMD" in module_options:
+            self.cmd = module_options["CMD"]
+        
+        if "USER" in module_options:
+            self.user = module_options["USER"]
+       
+    name = "schtask"
+    description = "Execute a scheduled task on the remote system"
+    supported_protocols = ["smb"]
+    opsec_safe = True
+    multiple_hosts = False
+    
+    def on_admin_login(self, context, connection):
+        self.logger = context.log
+        if self.cmd is None:
+            self.logger.fail("You need to specify a CMD to run")
+            return 1
+        if self.user is None:
+            self.logger .fail("You need to specify a USER to run the command as") 
+            return 1
+
+        self.logger.display("Connecting to the remote atsvc endpoint")
+        try:
+            exec_method = TSCH_EXEC(
+                connection.host if not connection.kerberos else connection.hostname + "." + connection.domain,
+                connection.smb_share_name,
+                connection.username,
+                connection.password,
+                connection.domain,
+                self.user,
+                self.cmd,
+                connection.kerberos,
+                connection.aesKey,
+                connection.kdcHost,
+                connection.hash,
+                self.logger,
+                connection.args.get_output_tries,
+                "C$" # This one shouldn't be hardcoded but I don't know where to retrive the info
+            )
+
+            output = exec_method.execute(self.cmd, True)
+            try:
+                if not isinstance(output, str):
+                    output = output.decode(connection.args.codec)
+            except UnicodeDecodeError:
+                self.logger.debug("Decoding error detected, consider running chcp.com at the target, map the result with https://docs.python.org/3/library/codecs.html#standard-encodings")
+                output = output.decode("cp437")
```

Comment:
Do we really need to specify this codec ?

Reply:
Yup, if you enable the debug mode you'll see that it always enters the exception and decode as cp437. We can remove the line but in that case we'll have to specify the cp437 codec in line 66. I don't know what could break tho..